dist/channel/index.js

```diff
@@ -21390,15 +21390,27 @@ const glob = __importStar(__nccwpck_require__(8090));
 class Artifact {
     uploadLogs() {
         return __awaiter(this, void 0, void 0, function* () {
-            const basePath = '/home/runner/snap/charmcraft/common/cache/charmcraft/log';
-            if (!fs.existsSync(basePath)) {
-                return 'No charmcraft logs generated, skipping artifact upload.';
-            }
-            const globber = yield glob.create(`${basePath}/*.log`);
-            const files = yield globber.glob();
-            const artifacts = artifact.create();
-            const result = yield artifacts.uploadArtifact('charmcraft-logs', files, basePath);
-            return `Artifact upload result: ${JSON.stringify(result)}`;
+            // We're running some charmcraft commands as sudo as others as a
+            // regular user - we want to capture both.
+            const basePaths = [
+                '/home/runner/snap/charmcraft/common/cache/charmcraft/log',
+                '/root/snap/charmcraft/common/cache/charmcraft/log/',
```

Comment:
I need to confirm, but I suspect this will fail because it's trying to pull from a sub-directory of `/root/`. I think I may need to shell out and copy these logs with `sudo` to somewhere the regular user can access.